api/api.py

```diff
@@ -1,7 +1,53 @@
-from api import app
-from flask import jsonify
+import hmac
+import hashlib
+
+from api import app, db, models
+from flask import jsonify, request
+
+
+def verify_hmac_hash(data, signature):
+    mac = hmac.new(app.config.get("GITHUB_SECRET").encode('utf-8'), data.encode('utf-8'), hashlib.sha256)
+    return hmac.compare_digest('sha256=' + mac.hexdigest(), str(signature))
 
 
 @app.route("/status")
 def status():
     return jsonify(status="ok")
+
+
+@app.route("/github-webhook", methods=["POST"])
+def github_webhook():
+    signature = request.headers.get("X-Hub-Signature-256")
+    data = request.get_data(as_text=True)
+
+    if verify_hmac_hash(data, signature):
+        if request.headers.get("X-Github-Event") == "ping":
+            return jsonify(msg="ok")
+        if request.headers.get("X-Github-Event") == "pull_request":
```

Comment:
Just curious if it makes any sense to use the `push` event, and check the ref of the branch? Was just thinking of an edge-case where people might commit directly to the main/master branch.

Otherwise, is it possible to get and compare the target branch of the PR? Again another potential edge-case where PRs might be opened into a feature branch, other than the service-defined branch in our config?